tests/lustre/typechecking/shouldPass/test_lustre_v4.lus

```diff
@@ -0,0 +1,352 @@
+---------------------------------------------------------------
+-- Examples from the V4 reference manual
+---------------------------------------------------------------
+
+-- Section 1
+
+node count (x, y: int) returns (s: int);
+let
+  s = 2 * (x+y);
+tel
+
+node EDGE (X: bool) returns (Y: bool);
+let
+  Y = false -> X and not pre(X);
+tel
+
+node FALLING_EDGE (X: bool) returns (Y: bool);
+let
+  Y = EDGE(not X);
+tel
+
+node SWITCH1 (set, reset, init: bool) returns (level: bool);
+let
+  level = init -> if set then true else if reset then false else pre(level);
+  assert not (set and reset);
+tel
+
+
+node SWITCH (set, reset, init: bool) returns (level: bool);
+let
+  level = init -> if set and not pre(level) then true else if reset then false else pre(level);
+tel
+
+node x1 (X: bool; N: int) returns ();
+var
+  N1, PN, N2, N3, N4: int;
+let
+  N1 = 0 -> if X then pre N + 1 else pre N;
+  PN = 0 -> pre N;
+  N2 = if X then PN + 1 else PN;
+  N3 = if X then (0 -> pre N) + 1 else (0 -> pre N);
```

Comment:
This test fails. Error message seems to be wrong:

> line 41 col. 2: Failure cause: Type error: Expected type int but found type [int, int, int]->[int, real]

Reply:
Not sure why this fails. Taking a look at this.

Reply:
This was because of name clashing. `node N` and `N` also being used as a local variable